packages/lib/src/interactions/AxialSelectToZoom.tsx

```diff
@@ -17,18 +16,18 @@ interface Props extends CommonInteractionProps {
 function AxialSelectToZoom(props: Props) {
   const { axis, modifierKey, disabled } = props;
 
-  const { dataToWorld } = useAxisSystemContext();
+  const context = useAxisSystemContext();
   const moveCameraTo = useMoveCameraTo();
 
   const { width, height } = useThree((state) => state.size);
   const camera = useThree((state) => state.camera);
 
-  const { xVisibleDomain, yVisibleDomain } = useCameraState(
-    getVisibleDomains,
-    []
-  );
```

Comment:
We don't actually need to re-render this component when the visible domains change. We just need to compute the visible domains when we actually need them - i.e. when drawing a selection rectangle (in the render prop of `SelectionTool`) and when completing a selection (in `onSelectionEnd`).